src/managers/RulesManager.ts

```diff
@@ -116,6 +124,14 @@ class RulesManager extends Map<string, InstanceType<typeof NimbleBaseRule>> {
 		return rule;
 	}
 
+	static hasRuleOfType(item: NimbleBaseItem, type: string) {
+		return item.system.rules.some((r) => r.type === type);
+	}
+
+	static getRuleOfType(item: NimbleBaseItem, type: string) {
+		return item.system.rules.find((r) => r.type === type);
+	}
+
```

Comment:
These don't need to be static, the reason for add and delete being static is so that migration runners and modules can easily create and delete rules while adhering to defined schema auto population